src/server/scripts/DragonIsles/AzureVault/boss_leymor.cpp

```diff
@@ -0,0 +1,581 @@
+    {
+        JustAppeared();
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.ScheduleEvent(EVENT_ERRATIC_GROWTH, 22s);
+        _events.ScheduleEvent(EVENT_WILD_ERUPTION, 12s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_ERRATIC_GROWTH:
+                    DoCastAOE(SPELL_ERRATIC_GROWTH_CHANNEL);
+                    _events.ScheduleEvent(EVENT_ERRATIC_GROWTH, 22s);
+                    break;
+                case EVENT_WILD_ERUPTION:
+                    DoCastAOE(SPELL_WILD_ERUPTION);
+                    _events.ScheduleEvent(EVENT_WILD_ERUPTION, 12s);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    EventMap _events;
+};
+
+struct npc_ley_line_sprouts : public ScriptedAI
+{
+    npc_ley_line_sprouts(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_LEY_LINE_SPROUT_AT);
+
+        DoCastAOE(SPELL_ARCANE_POWER, true);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        if (GetDifficulty() == DIFFICULTY_MYTHIC || GetDifficulty() == DIFFICULTY_MYTHIC_KEYSTONE)
+            me->CastSpell(me->GetPosition(), SPELL_VOLATILE_SAPLING, true);
+
+        if (TempSummon* tempSummon = me->ToTempSummon())
+        {
+            if (Unit* summoner = tempSummon->GetSummonerUnit())
+            {
+                if (Aura* aura = summoner->GetAura(SPELL_ARCANE_POWER))
+                    aura->ModStackAmount(-1);
+            }
+        }
+    }
+
+    void JustSummoned(Creature* summon) override
+    {
+        InstanceScript* instance = me->GetInstanceScript();
+        if (!instance)
+            return;
+
+        Creature* leymor = instance->GetCreature(DATA_LEYMOR);
+        if (!leymor)
+            return;
+
+        if (!leymor->IsAIEnabled())
+            return;
+
+        leymor->AI()->JustSummoned(summon);
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        if (spellInfo->Id != SPELL_EXPLOSIVE_BRAND_DAMAGE && spellInfo->Id != SPELL_ERUPTING_FISSURE_SPROUT_SELECTOR)
+            return;
+
+        me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SPROUT_DEATH, 0, 0);
+        me->KillSelf();
+    }
+};
+
+struct npc_volatile_sapling : public ScriptedAI
+{
+    npc_volatile_sapling(Creature* creature) : ScriptedAI(creature), _castedSappyBurst(false) { }
+
+    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/)
+    {
+        if (me->GetHealth() <= damage)
```

Comment:
HealthBelowPctDamaged is better

Reply:
Does it work correctly due to scaling? @mdX7

Reply:
What is even your question here @Nyr97

Reply:
HealthBelowPctDamaged does not work correctly due to scaling being half-baked into the core yet. That's why there are cases in which it is not recommended to be used.

Reply:
Wdym, HealthBelowPctDamaged is just using current health and CountHealthFromPct why should it care about scaling

Reply:
Ask @mdX7, he'll explain correctly.

Reply:
> HealthBelowPctDamaged is better

Why would unnecessary calculations be better than simply checking for >=?
The spell is cast on death essentially. Just that the mob casts it at 1hp and kills itself after spell cast is done. We dont really have to check for % in this case.

Ref scaled hp: damage being passed to the hook has some issues, its unscaled iirc